source/uwp/Renderer/lib/XamlHelpers.cpp

```diff
@@ -644,84 +644,71 @@ namespace AdaptiveNamespace::XamlHelpers
+        ABI::Windows::UI::Color color;
+        RETURN_IF_FAILED(GetColorFromAdaptiveColor(hostConfig, textColor, ContainerStyle_Default, false, false, &color));
 
-        // Set the color to Attention color
-        ComPtr<IAdaptiveHostConfig> hostConfig;
-        RETURN_IF_FAILED(renderContext->get_HostConfig(&hostConfig));
+        ComPtr<ABI::Windows::UI::Xaml::Documents::IRun> xamlLabelRun(labelRun);
 
-        ABI::Windows::UI::Color attentionColor;
-        RETURN_IF_FAILED(
-            GetColorFromAdaptiveColor(hostConfig.Get(), ForegroundColor_Attention, ContainerStyle_Default, false, false, &attentionColor));
+        ComPtr<ABI::Windows::UI::Xaml::Documents::ITextElement> labelRunAsTextElement;
+        RETURN_IF_FAILED(xamlLabelRun.As(&labelRunAsTextElement));
 
-        ComPtr<ABI::Windows::UI::Xaml::Documents::ITextElement> starRunAsTextElement;
-        RETURN_IF_FAILED(starRun.As(&starRunAsTextElement));
+        RETURN_IF_FAILED(labelRunAsTextElement->put_Foreground(XamlHelpers::GetSolidColorBrush(color).Get()));
 
-        RETURN_IF_FAILED(starRunAsTextElement->put_Foreground(XamlHelpers::GetSolidColorBrush(attentionColor).Get()));
+        ABI::AdaptiveNamespace::TextSize textSize;
+        RETURN_IF_FAILED(inputLabelConfig->get_Size(&textSize));
 
-        ComPtr<ABI::Windows::UI::Xaml::Documents::IInline> starRunAsInline;
-        RETURN_IF_FAILED(starRun.As(&starRunAsInline));
+        UINT32 resultSize{};
+        RETURN_IF_FAILED(GetFontSizeFromFontType(hostConfig, ABI::AdaptiveNamespace::FontType_Default, textSize, &resultSize));
 
-        RETURN_IF_FAILED(xamlInlines->Append(starRunAsInline.Get()));
+        RETURN_IF_FAILED(labelRunAsTextElement->put_FontSize(resultSize));
 
         return S_OK;
     }
 
-    HRESULT FormatLabelWithHostConfig(_In_ ABI::AdaptiveNamespace::IAdaptiveRenderContext* renderContext,
-                                      _In_ boolean isRequired,
-                                      ABI::Windows::UI::Xaml::Documents::IRun* labelRun)
+    // TextBlock and RichTextBlock xaml types both support inlines, so we reuse the same method for rendering the " *" hint in both cases
+    HRESULT AddRequiredHintInline(_In_ ABI::AdaptiveNamespace::IAdaptiveHostConfig* hostConfig,
+                                  _In_ ABI::AdaptiveNamespace::IAdaptiveInputLabelConfig* inputLabelConfig,
+                                  IVector<ABI::Windows::UI::Xaml::Documents::Inline*>* inlines)
     {
-        ComPtr<IAdaptiveHostConfig> hostConfig;
-        RETURN_IF_FAILED(renderContext->get_HostConfig(&hostConfig));
+        ComPtr<IVector<ABI::Windows::UI::Xaml::Documents::Inline*>> xamlInlines(inlines);
 
-        ComPtr<IAdaptiveInputsConfig> inputsConfig;
-        RETURN_IF_FAILED(hostConfig->get_Inputs(&inputsConfig));
+        // Create an inline for the suffix
+        ComPtr<IRun> starRun = XamlHelpers::CreateXamlClass<IRun>(HStringReference(RuntimeClass_Windows_UI_Xaml_Documents_Run));
```

Comment:
>starRun [](start = 21, length = 7)

Should we call this something different since it may not be a star? requiredHintRun?